spec/requests/articles/update_article_spec.rb

```diff
@@ -6,12 +6,75 @@
     @user1    = create(:user)
     @user2    = create(:user)
     @article1 = create(:article, status: "available", user: @user1)
+    @article2 = create(:article, status: "unavailable", user: @user1)
     @headers  = {
       'CONTENT_TYPE': 'application/json',
       'ACCEPT': 'application/json'
     }
   end
 
+  it 'errors on bad/missing id/user_id' do
+    expect(@article1[:user_id]).to eq(@user1[:id])
+
+    payload = {
+      "query": "mutation updateArticle ($article: UpdateArticleInput!) { 
+        updateArticle (input: $article) { 
+          article {
+            id 
+            name 
+            user { id name } 
+          }
+          errors
+        } 
+      }",
+      "variables": { "article": { "id": "1000", "userId": "1000" } }
+    }
+
+    post '/graphql', headers: @headers, params: JSON.generate(payload)
+
+    article = JSON.parse(response.body, symbolize_names: true).dig(:data, :updateArticle)
+
+    expect(response).to be_successful
+    expect(article).to have_key(:article)
+    expect(article[:article]).to eq(nil)
+    expect(article).to have_key(:errors)
+    expect(article[:errors]).to be_a(Array)
+    expect(article.dig(:errors, 0)).to be_a(String)
+    expect(article.dig(:errors, 0)).to eq("unable to find user")
```

Comment:
Typically, if you are asserting on the actual value of a piece of data, you don't also need to assert on its data type. For example, if this data were actually an Integer, it would be caught as a failure by the assertion on its actual value, so asserting that it were a string isn't also necessary. This is more of a nit but it can clean up your tests a bit